packages/agent/src/agent/http/index.ts

```diff
@@ -245,19 +253,21 @@ other computations so that this class can stay as simple as possible while
 allowing extensions.
  */
 export class HttpAgent implements Agent {
-  public rootKey: ArrayBuffer;
```

Comment:
I've removed the public `rootKey` property in favor of the async `getRootKey`, since we cannot guarantee availability of the correct root key when `fetchRootKey` is set to true.